### PR TITLE
Issue #21

### PR DESCRIPTION
**Pull request by kikogolk**
Fixing issue #21

### CODE REVIEW

File: restapi/src/main/java/com/craftsmanship/tfm/restapi/controllers/OrderRestController.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+        try {
+            Order orderResponse = orderPersistence.create(order);
+
+            LOGGER.info("REST API orderResponse = " + orderResponse);
+
+            // Send messages to Kafka topic
+            this.sendItemOperations(OperationType.CREATED, orderResponse.getItemPurchases());
+            return orderResponse;
+        } catch (ItemDoesNotExist e) {
```
Comment (carlosTorres07) [inline restapi/src/main/java/com/craftsmanship/tfm/restapi/controllers/OrderRestController.java:60]: I do not get when this exception occur. At the end is thrown by OrderPersistencGrpcClient class, but I do not get it.

File: restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java [hunk 2]
```diff
@@ -24,7 +25,12 @@
   private int serverPort;
 
   @Bean
-  public ItemPersistenceGrpc persistenceHandler() {
+  public ItemPersistenceGrpc itemPersistenceHandler() {
```
Comment (carlosTorres07) [inline restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java:28]: Would not be better ItemPersistence ?

File: restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java [hunk 3]
```diff
@@ -24,7 +25,12 @@
...
     return new ItemPersistenceGrpc(serverHost, serverPort);
   }
+
+  @Bean
+  public OrderPersistenceGrpc orderPersistenceHandler() {
```
Comment (carlosTorres07) [inline restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java:33]: Would not be better OrderPersistence ?

File: restapi/src/main/java/com/craftsmanship/tfm/restapi/controllers/OrderRestController.java (hunk 1 above)
Comment (kikogolk) [reply restapi/src/main/java/com/craftsmanship/tfm/restapi/controllers/OrderRestController.java:60]: The creation of an Order should fail if it contains an item that does not already exist in persistence.

File: restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java (hunk 2 above)
Comment (kikogolk) [reply restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java:28]: I wanted to let clear that in production ItemPersistenceGrpc will be used. But I can change if you want.

File: restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java (hunk 3 above)
Comment (kikogolk) [reply restapi/src/main/java/com/craftsmanship/tfm/restapi/config/PersistenceConfig.java:33]: I wanted to let clear that in production OrderPersistenceGrpc will be used. But I can change if you want.

**Review by carlosTorres07 — APPROVED** [review]

**Pull request merged**